### ISSUE TITLE
EarlGrey 2.0 | Exception when checking for Accessibility ID 

### ISSUE THREAD

**Issue by mitchell-asb** [google/EarlGrey#781]
I have an accessibilityIdentifier that is a number and its failing on the `isEqualToString` selector as `__NSCFNumber` doesnt have that selector. I will submit a pr adding a check if the class instance supports the selector `isEqualToString` unless anyone else has a better idea (maybe a cast from number to string?)
```
Uncaught exception in Application: -[__NSCFNumber isEqualToString:]: unrecognized selector sent to instance 0xd589f038d3507a35
0   CoreFoundation                      0x000000010ac0b1bb __exceptionPreprocess + 331
1   libobjc.A.dylib                     0x000000010980a735 objc_exception_throw + 48
2   CoreFoundation                      0x000000010ac29f44 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3   CoreFoundation                      0x000000010ac0fed6 ___forwarding___ + 1446
4   CoreFoundation                      0x000000010ac11da8 _CF_forwarding_prep_0 + 120
5   AppFramework                        0x0000000105548a24 __42+[GREYMatchers matcherForAccessibilityID:]_block_invoke + 388
6   AppFramework                        0x0000000105536868 -[GREYElementMatcherBlock matches:] + 72
7   AppFramework                        0x0000000105523425 -[GREYBaseMatcher matches:describingMismatchTo:] + 85
8   AppFramework                        0x0000000105558377 -[GREYAllOf matches:describingMismatchTo:] + 471
9   AppFramework                        0x0000000105558127 -[GREYAllOf matches:] + 167
10  AppFramework                        0x0000000105560c3c -[GREYElementFinder elementsMatchedInProvider:] + 1260
11  AppFramework                        0x0000000105565ed2 __78-[GREYElementInteraction matchElementsWithTimeout:syncBeforeMatch:matchBlock:]_block_invoke + 322
12  AppFramework                        0x00000001055139b4 __59-[GREYUIThreadExecutor executeSyncWithTimeout:block:error:]_block_invoke + 68
13  AppFramework                        0x00000001054fe4ca __98-[GREYRunLoopSpinner grey_drainRunLoopInActiveModeAndCheckCondition:explicitRunLoopDrain:forTime:]_block_invoke + 250
14  AppFramework                        0x00000001054ff8bf __110-[GREYRunLoopSpinner grey_setupObserverInMode:withBeforeSourcesBlock:beforeWaitingBlock:explicitRunLoopDrain:]_block_invoke.72 + 63
15  CoreFoundation                      0x000000010ab6fc27 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
16  CoreFoundation                      0x000000010ab6a0be __CFRunLoopDoObservers + 430
17  CoreFoundation                      0x000000010ab6a623 __CFRunLoopRun + 1235
18  CoreFoundation                      0x000000010ab69e11 CFRunLoopRunSpecific + 625
19  GraphicsServices                    0x000000010ea301dd GSEventRunModal + 62
20  UIKitCore                           0x0000000117d3181d UIApplicationMain + 140
21  ASBMobile_Dev                       0x000000010328fe50 main + 112
22  libdyld.dylib                       0x000000010c766575 start + 1```

**Comment by mitchell-johnson** [comment]
I discovered my issue. I had accidentally added an `accessibilityIdentifier` as a BOOL in an element on the affected page. although it's not valid to have a non string `accessibilityIdentifier` there is nothing to stop you doing it accidentally or deliberately. So I wonder if it's worth keeping this check in the PR

**Comment by tirodkar** [comment]
The check shouldn't be needed, especially since it would prevent you from finding actual issues like the `BOOL` you were returning. Thanks for bringing this up @mitchell-johnson. Closing the issue for now. Feel free to re-open it in case you see anything else that's related.
